app/src/main/java/org/simple/clinic/drugstockreminders/DrugStockWorker.kt

```diff
@@ -0,0 +1,136 @@
+import org.simple.clinic.setup.SetupActivity
+import org.simple.clinic.util.UtcClock
+import java.text.DateFormat
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.Date
+import java.util.Locale
+import java.util.Optional
+import javax.inject.Inject
+
+class DrugStockWorker(
+    private val context: Context,
+    workerParams: WorkerParameters
+) : RxWorker(context, workerParams) {
+
+  companion object {
+    private const val NOTIFICATION_CHANNEL_ID = "org.simple.clinic.drugstockreminders"
+    private const val NOTIFICATION_CHANNEL_NAME = "Drug Stock Updates"
+    private const val NOTIFICATION_ID = 5
+  }
+
+  @Inject
+  lateinit var clock: UtcClock
+
+  @Inject
+  lateinit var drugStockReminder: DrugStockReminder
+
+  @Inject
+  @TypedPreference(UpdateDrugStockReportsMonth)
+  lateinit var updateDrugStockReportsMonth: Preference<Optional<String>>
+
+  private val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+  init {
+    ClinicApp.appComponent.inject(this)
+  }
+
+  override fun createWork(): Single<Result> {
+    createNotificationChannel()
+
+    val formattedDate = previousMonthFormattedString()
+
+    return drugStockReminder
+        .reminderForDrugStock(formattedDate)
+        .map { response ->
+          when (response) {
+            is Found -> drugStockReportFound(response.drugStockReminderResponse.month)
+            NotFound -> drugStockReportNotFound(formattedDate)
+          }
+        }
+  }
+
+  private fun previousMonthFormattedString(): String {
+    val dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-DD", Locale.ENGLISH)
+    val previousMonthsDate = LocalDateTime.now(clock).minusMonths(1)
+    return dateFormat.format(previousMonthsDate)
+  }
+
+  private fun drugStockReportNotFound(previousMonthsDate: String): Result {
+    updateDrugStockReportsMonth.set(Optional.of(previousMonthsDate))
+    return Result.failure()
+  }
+
+  private fun drugStockReportFound(currentMonthsDate: String): Result {
+    updateDrugStockReportsMonth.set(Optional.of(currentMonthsDate))
+    notificationManager.notify(NOTIFICATION_ID, drugStockReminderNotification(currentMonthsDate))
+    return Result.success()
+  }
+
+  private fun drugStockReminderNotification(currentMonthsDate: String): Notification {
+    val monthAndYear = formatDateForNotification(currentMonthsDate)
+    val notificationHeaderText = context.getString(R.string.drug_stock_reminder_notification, monthAndYear)
+
+    return NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID)
+        .setSmallIcon(R.drawable.ic_app_update_notification_logo)
+        .setContentTitle(context.getString(R.string.app_name))
+        .setTicker(context.getString(R.string.app_name))
+        .setPriority(NotificationCompat.PRIORITY_HIGH)
+        .setStyle(NotificationCompat.BigTextStyle().bigText(notificationHeaderText))
```

Comment:
We should add `setContentText` as well for `notificationHeaderText` as if we only add `setStyle(NotificationCompat.BigTextStyle().bigText(notificationHeaderText))` when we look at the notification it would only render the title and we will have to hold and swipe down to look at the content text.